src/newrelic.ts

```diff
@@ -40,53 +65,361 @@ function addSegment(spec: Spec) {
+          telemetryCollector.updateSpecList(specMetadata);
+      }
+  })
+
+  seneca.order.outward.add((spec: Spec) => {
+      if (segmentIsEnabled) {
+          changeSpec(spec, 'outward').endSegments()
+      }
+      if (telemetryCollector) {
+          const specMetadata = telemetryCollector.extractFromSpec(spec, 'outward');
+          telemetryCollector.updateSpecList(specMetadata);
+      }
+  })
+
+  if (metricsIsEnabled) {
+    if (!options.metrics?.accountApiKey) {
+      throw new Error("Please provide accountApiKey parameter to Metrics API");
+    }
+    const metricsClient = new MetricClient({
+      apiKey: options.metrics.accountApiKey,
+    })
 
-function newrelic(this: any, _options: any) {
-  // const seneca: any = this
+    this.metricsClient = metricsClient;
+  }
 }
 
+function newrelic(this: any, options: NewRelicOptions) {
+    const seneca: any = this
+
+    seneca
+      .message('plugin:newrelic,get:info', get_info)
+      .message('plugin:newrelic,api:metrics', metric_handler)
+      .message('plugin:newrelic,api:metrics', metrics_validation)
+
+    async function get_info(this: any, _msg: any) {
+      return {
+        ok: true,
+        name: 'newrelic',
+        details: {
+          sdk: 'newrelic'
+        }
+      }
+    }
+
+    const _metricBatchBuilder = (metric: MetricBase<any>) => {
+      const batch = new MetricBatch({}, Date.now(), 1000);
+      batch.addMetric(metric)
+      return batch;
+    };
+
+    const _sendBatch = (batch: MetricBatch) => {
+      seneca.metricsClient.send(batch, (err: any, res: any, body: any) => {
+        // TODO: Implement some logging to handle failed batches
+        if (err) {
+          console.log('aq', err);
+        }
+        console.log(res.statusCode);
+        console.log(body);
+      })
+    }
+
+    async function metric_count_handler(name: string, value: number, attributes?: any) {
+      const countMetrics = new CountMetric(name, value, attributes || {});
+      const batch = _metricBatchBuilder(countMetrics);
+      _sendBatch(batch);
+    }
+
+    async function metric_gauge_handler(name: string, value: number, attributes?: any) {
+      const gaugeMetric = new GaugeMetric(name, value, attributes || {});
+      const batch = _metricBatchBuilder(gaugeMetric);
+      _sendBatch(batch);
+    }
+
+    async function metric_summary_handler(name: string, msg: any, attributes?: any) {
+      const extractSummaryData = (msg: any): SummaryValue => {
+        const { count, sum, min, max } = msg;
+        return {
+          count: count || 0,
+          sum: sum || 0,
+          min: min || Infinity,
+          max: max || -Infinity ,
+        };
+      }
+      const summaryData = extractSummaryData(msg);
+      const summaryMetrics = new SummaryMetric(name, summaryData, attributes || {});
+      const batch = _metricBatchBuilder(summaryMetrics);
+      _sendBatch(batch);
+    }
+
+    async function metrics_validation(this: any, msg: any) {
```

Comment:
there is an easier way to do this - not documented yet, but you can still try it out: https://github.com/rjrodger/gubu

example: https://github.com/senecajs/seneca-flow/blob/main/src/flow.ts#L85

please do give me feedback on this as it is a very new feature